cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ListenersUtils.java

```diff
@@ -699,4 +699,50 @@ public static String advertisedPortFromOverrideOrParameter(GenericKafkaListener
 
         return String.valueOf(advertisedPort != null ? advertisedPort : port);
     }
+/**
+     * Finds bootstrap service external IPs
+     *
+     * @throws  UnsupportedOperationException Throws UnsupportedOperationException if called for other KafkaListenerType.
+     * 
+     * @param listener  Listener for which the external IPs should be found
+     * @return          External IPs or null if not specified
+     */
+    public static List<String> bootstrapExternalIPs(GenericKafkaListener listener){
+		if (KafkaListenerType.NODEPORT == listener.getType()) {
+			if (listener.getConfiguration() != null
+	                && listener.getConfiguration().getBootstrap() != null) {
+	            return listener.getConfiguration().getBootstrap().getExternalIPs();
+	        } else {
+	            return null;
+	        }
+		}else {
+			throw new UnsupportedOperationException("Bootstrap externalIP are not used with listerner type "+ listener.getType());
+		}
+    }
+
+    /**
+     * Finds broker service external IPs
+     *
+     * @throws  UnsupportedOperationException Throws UnsupportedOperationException if called for other KafkaListenerType.
+     *
+     * @param listener  Listener for which the external IPs should be found
+     * @param pod       Pod ID for which we should get the configuration option
+     * @return          External IPs or null if not specified
+     */
+    public static List<String> brokerExternalIPs(GenericKafkaListener listener, int pod){
+		 if (KafkaListenerType.NODEPORT == listener.getType()) {
+			 if (listener.getConfiguration() != null
+		                && listener.getConfiguration().getBrokers() != null) {
+		            return listener.getConfiguration().getBrokers().stream()
+		                    .filter(broker -> broker != null && broker.getBroker() != null && broker.getBroker() == pod && broker.getExternalIPs() != null)
+		                    .map(GenericKafkaListenerConfigurationBroker::getExternalIPs)
+		                    .findAny()
+		                    .orElse(null);
+		        } else {
+		            return null;
+		        }
+		 }else {
+			 throw new UnsupportedOperationException("Broker externalIP are not used with listerner type "+ listener.getType());
+		 }
+    }
```

Comment:
You should mirror what the other methods do => just return here and do validation in the validation class (`ListenersValidator` or something like that).